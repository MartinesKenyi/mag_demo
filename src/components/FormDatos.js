import React, { useContext, useState } from 'react';
import { SuscriptionContext } from '../suscription/SuscriptionContext';
import { UsuarioContext } from '../usuario/UsuarioContext';
import { useForm } from '../hooks/useForm';
import { types } from '../types/types';

export const FormDatos = ({ history }) => {

    const { suscriptionState } = useContext(SuscriptionContext);
    const { dispatch } = useContext(UsuarioContext);
    const { precio } = suscriptionState.sucriptionActive;

    const [error, setError] = useState('')

    const [valueForm, handleInputChange] = useForm({
        usuario: '',
        card: '',
        fexpira: '',
        cvc: ''
    });

    const { usuario, card, fexpira, cvc } = valueForm;

    const handlePagar = () => {

        if (validForm()) {
            dispatch({
                type: types.setUsuario,
                payload: {
                    usuario, 
                    card, 
                    fexpira, 
                    cvc
                }
            });
            history.push("/confirmacion");
        }
    }
    const validForm = () => {
        if(usuario.length < 5) {
            console.log("Digite su nombre y apellidos!")
            setError("Digite su nombre y apellidos!")
            return false;
        }
        if(card.length < 12) {
            console.log("Digite la tarjeta correctamente!")
            setError("Digite la tarjeta correctamente!")
            return false;
        }
        if(fexpira.length < 5) {
            console.log("Digite la fecha de F. expiración correctamente!")
            setError("Digite la fecha de F. expiración correctamente!")
            return false;
        }
        if(cvc.length < 3) {
            console.log("Digite CVC correctamente!")
            setError("Digite CVC correctamente!")
            return false;
        }

        return true;
    }

    return (
        <>
            <h2
                className="datos__title-h2"
            >Registre sus datos</h2>

            <fieldset className="datos__fieldset">
                <label
                    className="datos__label"
                >
                    Nombre y apellidos
                </label>
                <input
                    type="text"
                    placeholder="Nombre y apellido"
                    autoComplete="off"
                    name="usuario"
                    className="datos__input"
                    value={usuario}
                    onChange={handleInputChange}
                />

            </fieldset>

            <fieldset className="datos__fieldset">
                <label
                    className="datos__label"
                >
                    Número de tarjeta
                </label>
                <input
                    type="number"
                    placeholder=".... .... .... ...."
                    autoComplete="off"
                    name="card"
                    className="datos__input"
                    value={card}
                    onChange={handleInputChange}
                />
            </fieldset>

            <div className="row">
                <div style={{ width: '50%', float: 'left' }}>
                    <fieldset className="datos__fieldset">
                        <label
                            className="datos__label"
                        >
                            F. Expira
                </label>
                        <input
                            type="number"
                            placeholder="MM/AA"
                            autoComplete="off"
                            name="fexpira"
                            className="datos__input"
                            style={{ width: '60%' }}
                            value={fexpira}
                            onChange={handleInputChange}
                        />
                    </fieldset>
                </div>
                <div style={{ width: '50%', float: 'left' }}>
                    <fieldset className="datos__fieldset">
                        <label
                            className="datos__label"
                        >
                            CVC
                </label>
                        <input
                            type="number"
                            autoComplete="off"
                            name="cvc"
                            className="datos__input"
                            style={{ width: '60%' }}
                            value={cvc}
                            onChange={handleInputChange}
                        />
                    </fieldset>
                </div>
            </div>
            <div style={{
                color: '#FD3D4C'
            }}>
                { error }
            </div>

            <div
                style={{ textAlign: 'center' }}
            >
                <button
                    className="button"
                    onClick={handlePagar}
                >
                    Pagar S/. {precio}.00
                </button>
            </div>
        </>
    )
}
