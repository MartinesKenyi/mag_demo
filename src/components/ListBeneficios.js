import React from 'react';
import PropTypes from 'prop-types';

export const ListBeneficios = ({ beneficio }) => {
    return (
        <>
            <li className="li plans__li">{beneficio}</li>
        </>
    )
}

ListBeneficios.propTypes = {
    beneficio: PropTypes.string.isRequired,
}