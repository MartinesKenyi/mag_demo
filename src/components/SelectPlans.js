import React from 'react'

export const SelectPlans = () => {

    const handleCambiarPlan = () => {
        console.log("cambiar plan")
    }

    return (
        <div style={{ padding: "30px 75px" }}>
            <h3 style={{fontSize: '12px', color: '#252525c5'}}>
                plan
                <span style={{fontSize: '20px', color: '#000'}}> Estandar</span>
            </h3>
            <h1 style={{ display: 'flex', fontSize: "50px", margin: "10px" }}>
                <span style={{ fontSize: "12px", margin: "10px" }}>
                    S/
                    </span>
                    29
                    <span style={{ fontSize: "10px", margin: '35px 0 0 5px' }}> / al mes</span>
            </h1>

            <p style={{ fontSize: "11px", margin: "0 5px" }}> Lorem ipsum dolor sit amet, consectetur adipiscing eli </p>

            <hr style={{ border: '1pt solid #6b6b6b62', margin: "15px 0" }} />

            <ul className="ul">
                <li className="li plans__li">Sed ut perspiciatis unde omnis iste natus. </li>
                <li className="li plans__li">But I must explain to you how all this mistaken. </li>
                <li className="li plans__li">Sed ut perspiciatis unde omnis iste natus error.</li>
            </ul>

            <div
                style={{ textAlign: 'center', marginTop: '15%' }}
            >
                <div
                    onClick={handleCambiarPlan}
                    className="datos__cambiar-plan"
                >
                    {"Cambiar a plan premiun >"}
                </div>
            </div>
        </div>
    )
}