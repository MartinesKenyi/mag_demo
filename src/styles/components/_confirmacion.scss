.confirmacion__container{
    background-color: rgba(255, 255, 255, 0.452);
    
    backdrop-filter: blur(1px);
    border: solid 1px rgba(248, 248, 248, 0.178);
    border-radius: 10px;

    width: 600px;
    margin: 0 auto;
    

    box-shadow: 0px 10px 20px 5px rgba(110, 110, 110, 0.3);
    
    // tapa tocas la capas hijas 
    overflow: hidden;
    transition:all .3s ease;

    @media screen and (max-width: 1300px) {
        width: 600px;
    }

    @media screen and (max-width: 750px) {
        width: 600px;
    }

    @media screen and (max-width: 650px) {
        width: 80%;
    }
}

.confirmacion__bienvenido{
    color: #333333;

    span {
        color: black;
        font-weight: bold;
        font-size: 17px;
    }
}
.confirmacion__plan{
    margin: 10px 0;
}
.confirmacion__precio{
    
    color: #333333;
}
.confirmacion__descripcion {
    color: #6D757D;
    margin: 20px 0;
}
.confirmacion__ref {
    text-decoration-line: underline; 
    color:#252525c5;
    cursor: pointer;
}
.confirmacion__img {
    
    height: 100%;
    width: 100%;
}

/*/*/
.success-checkmark {
    width: 80px;
    height: 115px;
    margin: 0 auto;
    
    .check-icon {
        width: 80px;
        height: 80px;
        position: relative;
        border-radius: 50%;
        box-sizing: content-box;
        border: 4px solid rgba(17, 17, 17, 0.733);
        
        &::before {
            top: 3px;
            left: -2px;
            width: 30px;
            transform-origin: 100% 50%;
            border-radius: 100px 0 0 100px;
        }
        
        &::after {
            top: 0;
            left: 30px;
            width: 60px;
            transform-origin: 0 50%;
            border-radius: 0 100px 100px 0;
            animation: rotate-circle 4.25s ease-in;
        }
        
        &::before, &::after {
            content: '';
            height: 100px;
            position: absolute;
            transform: rotate(-45deg);
        }
        
        .icon-line {
            height: 5px;
            background-color: rgb(46, 46, 46);;
            display: block;
            border-radius: 2px;
            position: absolute;
            z-index: 10;
            
            &.line-tip {
                top: 46px;
                left: 14px;
                width: 25px;
                transform: rotate(45deg);
                animation: icon-line-tip 0.75s;
            }
            
            &.line-long {
                top: 38px;
                right: 8px;
                width: 47px;
                transform: rotate(-45deg);
                animation: icon-line-long 0.75s;
            }
        }
        
        .icon-circle {
            top: -4px;
            left: -4px;
            z-index: 10;
            width: 80px;
            height: 80px;
            border-radius: 50%;
            position: absolute;
            box-sizing: content-box;
            border: 4px solid rgba(17, 17, 17, 0.904);
        }
        
        .icon-fix {
            top: 8px;
            width: 5px;
            left: 26px;
            z-index: 1;
            height: 85px;
            position: absolute;
            transform: rotate(-45deg);
        }
    }
}

@keyframes rotate-circle {
    0% {
        transform: rotate(-45deg);
    }
    5% {
        transform: rotate(-45deg);
    }
    12% {
        transform: rotate(-405deg);
    }
    100% {
        transform: rotate(-405deg);
    }
}

@keyframes icon-line-tip {
    0% {
        width: 0;
        left: 1px;
        top: 19px;
    }
    54% {
        width: 0;
        left: 1px;
        top: 19px;
    }
    70% {
        width: 50px;
        left: -8px;
        top: 37px;
    }
    84% {
        width: 17px;
        left: 21px;
        top: 48px;
    }
    100% {
        width: 25px;
        left: 14px;
        top: 45px;
    }
}

@keyframes icon-line-long {
    0% {
        width: 0;
        right: 46px;
        top: 54px;
    }
    65% {
        width: 0;
        right: 46px;
        top: 54px;
    }
    84% {
        width: 55px;
        right: 0px;
        top: 35px;
    }
    100% {
        width: 47px;
        right: 8px;
        top: 38px;
    }
}