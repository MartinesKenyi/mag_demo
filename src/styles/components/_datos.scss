.datos__container{
    background-color: rgba(255, 255, 255, 0.452);
    
    backdrop-filter: blur(1px);
    border: solid 1px rgba(248, 248, 248, 0.178);
    border-radius: 10px;

    width: 50%;
    margin: 0 auto;
    padding: 15px;

    box-shadow: 0px 10px 20px 5px rgba(110, 110, 110, 0.3);
    
    // tapa tocas la capas hijas 
    overflow: hidden;
    transition:all .9s ease;

    @media screen and (max-width: 1300px) {
        width: 70%;
    }

    @media screen and (max-width: 750px) {
        width: 80%;
    }
}

.datos__title-h2{
    font: bold 24px/29px "Haas Grot Text R Web", "Helvetica Neue", Helvetica, Arial, sans-serif;
    margin-bottom: 26px;
}

.datos__fieldset{
    margin: 0 0 12px;
    border: 0;
    line-height: 1;
    padding: 0;
}
.datos__label{
    display: block;
    font: bold 15px/24px "Haas Grot Text R Web", "Helvetica Neue", Helvetica, Arial, sans-serif;
    margin: 14px 0 4px;
    color: #0d0c22;
}
.datos__input {
    border: 1px solid transparent;
    background: #f3f3f4e5;
    color: #0d0c22;
    font-size: 14px;
    font-family: "Haas Grot Text R Web", "Helvetica Neue", Helvetica, Arial, sans-serif;
    padding: 10px 16px;
    outline: none;
    border-radius: 8px;
    width: 80%;
    transition:all .3s ease;
    
    &:hover {
        background-color: #FFF;
        border-color: rgba(61, 36, 36, 0.4);
        box-shadow: 0 0 0 4px rgb(61 36 36 / 10%);
    }
}

.datos__cambiar-plan{
    text-decoration-line: underline; 
    color:#252525c5;
    cursor: pointer;
}
.datos__select-plans{
    padding: 30px 75px;

    @media screen and (max-width: 1600px) {
        padding: 30px 50px;
    }

    @media screen and (max-width: 750px) {
        padding: 30px;
    }

    @media screen and (max-width: 650px) {
        padding: 30px 10px;
    }
}

.datos__cambiar{
    text-align: center;
    font-weight: bold;
}

.datos__modal {
    // display: none;
    position: fixed;
    z-index: 1;
    overflow: auto;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background: #2e3f5f86;

    display: flex;
    align-items: center;
    justify-content: center;

    animation-name: datos__modal;
    animation-duration: 1s;
}
@keyframes datos__modal {
    from{top:330px; opacity:0}
    to{top:0; opacity: 1;}
}