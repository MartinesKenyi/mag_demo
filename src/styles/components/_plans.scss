.plans__container{
    background-color: rgba(255, 255, 255, 0.466);
    
    backdrop-filter: blur(1px);
    border: solid 1px rgba(209, 209, 208, 0.05);
    border-radius: 10px;

    width: 20%;
    margin: 0 auto;
    padding: 15px;

    box-shadow: 0px 10px 20px 5px rgba(24, 24, 24, 0.3);
    
    // tapa tocas la capas hijas 
    overflow: hidden;
    transition:all .9s ease;

    &:hover {
        transform:scale(1.1);
    }

    @media screen and (max-width: 1300px) {
        width: 38%;
    }

    @media screen and (max-width: 750px) {
        width: 100%;
    }

}

.plans__popular {
    font-size: 10px;
    color: #FCFCFC;
    padding: 5px;
    border-radius: 5px;
    background: #575757;
}

.plans__plan {
    color: #FCFCFC;
    font-weight: bold;
}

.plans__li{
    font-size: 13px;
    color: #313131;
    margin: 5px;
}
.plans__div_button{
    width: 100%;
    position: fixed;
    bottom: 0;
    left: 0%;
    text-align: center;
}