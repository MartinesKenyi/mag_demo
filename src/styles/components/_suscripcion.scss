
.suscription__container_title {
    text-align: center; 
    padding-top: calc(100% - 95%);

    h1{
        color: #fcfcfc;
        font-weight: bold;
        margin: 10px;

        span {
            color: #131313;
            font-weight: 100;
            font-size: 20px;
        }
    }

    p{
        color: #6D757D;
    }

    @media screen and (max-width: 1300px) {
        padding-top: calc(100% - 89%);
    }

    @media screen and (max-width: 750px) {
        padding-top: calc(100% - 75%);
    }
}

.suscription__plans {
    display: flex;
    margin: 0 20%;
    padding-top: calc(100% - 95%);

    

    @media screen and (max-width: 750px) {
        margin: 0;
    }
    @media screen and (max-width: 650px) {
        overflow-x: auto;
    }
}
// .suscription__plans::-webkit-scrollbar {
//     width: 0;
// }