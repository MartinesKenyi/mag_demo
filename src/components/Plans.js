import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { SuscriptionContext } from '../suscription/SuscriptionContext';
import { types } from '../types/types';
import { ListBeneficios } from './ListBeneficios';

export const Plans = ({ history, tipo, precio, descricion, popular, beneficios }) => {

    const { dispatch } = useContext(SuscriptionContext);

    const handleSuscription = () => {
        dispatch({
            type: types.setSuscription,
            payload: {
                tipo, 
                precio, 
                descricion, 
                popular, 
                beneficios
            }
        });
        
        history.push("/datos");
    }

    return (
        <>
            <div className="plans__container">
                <div style={{ width: '100%', display: 'flex', justifyContent: 'flex-end' }} >
                    <div className="plans__popular" hidden={!popular}> POPULAR </div>
                </div>
                <h3 className="plans__plan">{tipo}</h3>
                <h1 className="plans__price">
                    <span style={{ fontSize: "12px", margin: "10px" }}>
                        S/
                    </span>
                    {precio}
                    <span style={{ fontSize: "10px", margin: '35px 0 0 5px' }}> / al mes</span>
                </h1>
                <p className="plans__description"> {descricion} </p>

                <hr className="hr" />

                <ul className="ul"
                    style={{ marginBottom: "60px" }}
                >
                    {
                        beneficios.map((beneficio, i) => (
                            <ListBeneficios
                                key={i}
                                beneficio={beneficio}
                            />
                        ))
                    }
                </ul>


                <div
                    className="plans__div_button"
                >

                    <button
                        onClick={handleSuscription}
                        className="button"
                    >
                        Suscribirme
                </button>
                </div>
            </div>
        </>
    )
}

Plans.propTypes = {
    tipo: PropTypes.string.isRequired, 
    precio: PropTypes.string.isRequired,
    descricion: PropTypes.string, 
    popular: PropTypes.bool,
    beneficios: PropTypes.array,
}