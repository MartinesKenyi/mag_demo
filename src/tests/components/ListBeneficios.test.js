import React from 'react';
import { shallow } from 'enzyme';
import { ListBeneficios } from '../../components/ListBeneficios';


describe('Pruebas en <ListBeneficios />', () => {
    
    test('Debe de mostrarse correctamente', () => {
       
        const wrapper = shallow (
            <ListBeneficios
                beneficio={"Hola"}
            />
        );

        expect(wrapper).toMatchSnapshot();
        
    });
    

})
