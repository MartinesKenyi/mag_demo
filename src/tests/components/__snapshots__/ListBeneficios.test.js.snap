// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ListBeneficios /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="li plans__li"
  >
    Hola
  </li>
</Fragment>
`;
